File: src/main/java/com/jcloisterzone/ai/player/DummyAiPlayer.java
package com.jcloisterzone.ai.player;

import java.util.Random;

import com.jcloisterzone.ai.AiPlayer;
import com.jcloisterzone.game.state.GameState;
import com.jcloisterzone.io.message.ReplayableMessage;

import io.vavr.collection.Vector;

public class DummyAiPlayer implements AiPlayer {

    private Random random = new Random();

    @Override
    public ReplayableMessage apply(GameState state) {
        Vector<ReplayableMessage> messages = getPossibleActions(state);
        return messages.get(random.nextInt(messages.length()));
    }
}

File: src/main/ts/com/jcloisterzone/ai/player/DummyAiPlayer.ts
import type { ReplayableMessage } from "../../io/message/ReplayableMessage.js";
import type { GameState } from "../../game/state/GameState.js";
import { AiPlayer } from "../AiPlayer.js";

/** Picks a random legal move. Port of `ai/player/DummyAiPlayer`. */
export class DummyAiPlayer extends AiPlayer {
  override apply(state: GameState): ReplayableMessage {
    const messages = this.getPossibleActions(state);
    return messages.get(Math.floor(Math.random() * messages.length()));
  }
}
